src/Tools/PerfDiff/Program.cs

```diff
@@ -18,48 +15,26 @@ namespace PerfDiff
     internal sealed class Program
     {
         internal const int UnhandledExceptionExitCode = 1;
-        private static ParseResult? s_parseResult;
 
-        private static async Task<int> Main(string[] args)
-        {
-            var rootCommand = DiffCommand.CreateCommandLineOptions();
-            rootCommand.Handler = CommandHandler.Create(new DiffCommand.Handler(RunAsync));
-
-            // Parse the incoming args so we can give warnings when deprecated options are used.
-            s_parseResult = rootCommand.Parse(args);
-
-            return await rootCommand.InvokeAsync(args).ConfigureAwait(false);
-        }
+        private static Task<int> Main(string[] args)
+            => DiffCommand.CreateCommandLineOptions().Parse(args).InvokeAsync();
 
         public static async Task<int> RunAsync(
             string baseline,
             string results,
             string? verbosity,
             bool failOnRegression,
-            IConsole console)
+            CancellationToken token)
         {
-            if (s_parseResult == null)
-            {
-                return 1;
-            }
-
             // Setup logging.
             var logLevel = GetLogLevel(verbosity);
-            var logger = SetupLogging(console, minimalLogLevel: logLevel, minimalErrorLevel: LogLevel.Warning);
-
-            // Hook so we can cancel and exit when ctrl+c is pressed.
-            var cancellationTokenSource = new CancellationTokenSource();
-            Console.CancelKeyPress += (sender, e) =>
```

Comment:
This is now being performed out of the box by System.CommandLine. All you need it to pass the cancellation token in SetAction to given async method